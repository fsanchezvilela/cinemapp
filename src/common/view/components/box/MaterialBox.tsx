import React from 'react';
import { styled } from '@material-ui/core/styles';
import {
  compose,
  spacing,
  palette,
  flexbox,
  display,
  borderRadius,
} from '@material-ui/system';

const Box = styled('div')(
  compose(spacing, palette, flexbox, display, borderRadius)
);

export interface IMaterialBox {
  children: React.ReactNode;
}

const MaterialBox = ({ children }: IMaterialBox) => (
  <Box
    display="flex"
    justifyContent="center"
    alignItems="center"
    flexDirection="column"
    color="white"
    bgcolor="#3f51b5"
    p={1}
    borderRadius="1rem"
  >
    {children}
  </Box>
);

export default MaterialBox;
