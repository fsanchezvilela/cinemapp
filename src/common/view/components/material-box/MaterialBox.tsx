import React from 'react';
import style from './MaterialBox.module.scss';

export interface IMaterialBox {
  children?: React.ReactNode;
}

const MaterialBox = ({ children }: IMaterialBox) => {
  return <div className={style['material-box']}>{children}</div>;
};

export default MaterialBox;
