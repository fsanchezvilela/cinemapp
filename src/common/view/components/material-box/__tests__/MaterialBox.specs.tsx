import React from 'react';
import { customRender } from '../../../../../../tests/test-utils';
import MaterialBox from '../MaterialBox';

describe('MaterialBox', () => {
  it('Render', () => {
    const rendered = customRender(<MaterialBox />);
    expect(rendered.container).toMatchSnapshot();
  });
  it('Should show text children', () => {
    const rendered = customRender(<MaterialBox>mock test</MaterialBox>);
    expect(rendered.getByText('mock test')).not.toBeNil();
  });
});
