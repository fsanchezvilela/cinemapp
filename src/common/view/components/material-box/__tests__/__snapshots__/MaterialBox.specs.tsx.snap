// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialBox Render 1`] = `
<div>
  <div
    class="material-box"
  />
</div>
`;
